### PR TITLE
Update clip loss calculation

### PR DESCRIPTION
**Pull request by sachinruk**
Hello, I'm the author of the blog you took the snippet from. I think this way of calculating is possibly slightly more accurate for calculation.

### CODE REVIEW

File: src/transformers/models/clip/modeling_clip.py [hunk 1]
```diff
@@ -61,14 +61,13 @@ def _expand_mask(mask: torch.Tensor, dtype: torch.dtype, tgt_len: Optional[int]
 
 # contrastive loss function, adapted from
 # https://sachinruk.github.io/blog/pytorch/pytorch%20lightning/loss%20function/gpu/2021/03/07/CLIP.html
-def contrastive_loss(logits: torch.Tensor, dim: int) -> torch.Tensor:
-    neg_ce = torch.diag(nn.functional.log_softmax(logits, dim=dim))
-    return -neg_ce.mean()
+def contrastive_loss(logits: torch.Tensor) -> torch.Tensor:
+    return nn.functional.cross_entropy(logits, torch.arange(len(logits)).to(logits.device))
```
Comment (patil-suraj) [inline src/transformers/models/clip/modeling_clip.py:65]:
```suggestion
    return nn.functional.cross_entropy(logits, torch.arange(len(logits), device=logits.device))
```

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for the PR!

And great blog post :)
